client/wayland/ibuswaylandim.c

```diff
@@ -1316,17 +1743,35 @@ registry_handle_global (void               *data,
         fprintf (priv->log, "wl_reistry gets interface: %s\n", interface);
         fflush (priv->log);
     }
-    if (!g_strcmp0 (interface, "zwp_input_method_v1")) {
-        priv->input_method =
-            wl_registry_bind (registry, name,
-                              &zwp_input_method_v1_interface, 1);
-        zwp_input_method_v1_add_listener (priv->input_method,
-                                          &input_method_listener, wlim);
-    }
-    if (!g_strcmp0 (interface, "zwp_input_panel_v1")) {
+    priv->serial = 0;
+    if (!g_strcmp0 (interface, zwp_input_method_manager_v2_interface.name)) {
+        priv->version = INPUT_METHOD_V2;
+        priv->input_method_manager_v2 =
+                wl_registry_bind (registry, name,
+                                  &zwp_input_method_manager_v2_interface, 1);
+    } else if (!g_strcmp0 (interface, "zwp_input_method_v1")) {
```

Comment:
For consistency it would probably be better to use `zwp_input_method_v1_interface.name` here.

Reply:
Thank you. You're right.